### PR TITLE
Refactor operator tests

### PR DESCRIPTION
**Pull request by ddmler**
Hey!

Type: code quality

Link to issue:

This pull request affects the following components **(please check boxes):**

* [ ] Core
* [ ] Analyzer
* [x] Compiler
* [ ] Control Flow Graph
* [ ] Documentation

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/ovr/phpsa/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change:

- removed duplicated tests
- created AbstractUnaryOp
- created AbstractBinaryOp
- fixed Bug: Notidentical was testing identical
- added support for bool type for plus and minus operators

Instead of copy&pasting the tests for every operator, we have now 1 test that checks every operator with the same data provider (all duplicates got eliminated). It checks every combination of supported types once (not some multiple times and others never like before)

still needed:
assert correct type? (currently commented out)
add every type to data provider

### CODE REVIEW

**Comment by ovr** [comment]
I need more time to review it, I will try it localy!

File: tests/PHPSA/Compiler/Expression/AbstractBinaryOp.php [hunk 1]
```diff
@@ -9,6 +9,17 @@
...
     /**
+     * Data provider for tests
+     */
+    public $data = [
+            CompiledExpression::INTEGER => 6,
+            CompiledExpression::DOUBLE => 2.5,
+            CompiledExpression::STRING => "test",
+            CompiledExpression::BOOLEAN => true,
+            CompiledExpression::NULL => null,
+        ];
```
Comment (ovr) [inline tests/PHPSA/Compiler/Expression/AbstractBinaryOp.php]: Indentation broken

**Comment by ddmler** [comment]
@ovr I fixed the indentation

**Comment by ovr** [comment]
going to close, reopen PR to restart tests

**Comment by ovr** [comment]
Thanks @ddmler 👍

**Pull request merged**